{"id":727,"status":"Fixed","summary":"Error trying to build try-catch block (AST)","labels":["Type-Defect","Priority-Medium"],"stars":0,"commentCount":3,"comments":[{"id":0,"commenterId":4086218909452486504,"content":"1. EXAMPLE\r\n\r\nNode nodeTry \u003d IR.block(\r\n  IR.var(\r\n    IR.name(\u0026quot;testing\u0026quot;), \r\n    IR.number(11)\r\n  )\r\n);\r\nNode nodeCatch \u003d IR.catchNode(\r\n  IR.name(\u0026quot;blabla\u0026quot;),\r\n    IR.block(\r\n      IR.var(\r\n        IR.name(\u0026quot;testing\u0026quot;),\r\n\tIR.number(22)\r\n      )\r\n    )\r\n);\r\nIR.tryCatch(nodeTry, nodeCatch);\r\n\r\n\r\n2. THROWS ERROR\r\n\r\nException in thread \u0026quot;main\u0026quot; java.lang.RuntimeException: java.lang.RuntimeException: INTERNAL COMPILER ERROR.\r\nPlease report this problem.\r\nnull\r\n  Node(FUNCTION tt): input.js:2:4\r\n    function tt() {\r\n  Parent(BLOCK): input.js:1:4\r\ntry {\r\n\r\n\r\n3. SOLUTION\r\n\r\nIR.block is verifing the catch node is a statement which it isnt.","timestamp":1336942779,"attachments":[]},{"id":1,"commenterId":-5060162230523776870,"content":"","timestamp":1337015384,"attachments":[]},{"id":2,"commenterId":1328304962299559429,"content":"This issue was closed by revision r1971.","timestamp":1337100551,"attachments":[]}]}